code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -309,6 +309,40 @@
 /obj/item/reagent_containers/spray/chemsprayer/bioterror
 	list_reagents = list("sodium_thiopental" = 100, "coniine" = 100, "venom" = 100, "condensedcapsaicin" = 100, "initropidril" = 100, "polonium" = 100)
 
+
+/obj/item/reagent_containers/spray/chemsprayer/janitor
+	name = "janitor chem sprayer"
+	desc = "A utility used to spray large amounts of cleaning reagents in a given area. It regenerates space cleaner by itself but it's unable to be fueled by normal means."
+	icon_state = "chemsprayer_janitor"
+	item_state = "chemsprayer_janitor"
+	lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
+	reagent_flags = NONE
+	list_reagents = list("cleaner" = 1000)
+	volume = 1000
+	amount_per_transfer_from_this = 5
+	var/generate_amount = 50
+	var/generate_type = "cleaner"
+	var/last_generate = 0
+	var/generate_delay = 10	//deciseconds
+
+/obj/item/reagent_containers/spray/chemsprayer/janitor/Initialize()
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/item/reagent_containers/spray/chemsprayer/janitor/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/obj/item/reagent_containers/spray/chemsprayer/janitor/process()
+	if(world.time < last_generate + generate_delay)
+		return
+	last_generate = world.time
+	generate_reagents()
+
+/obj/item/reagent_containers/spray/chemsprayer/janitor/proc/generate_reagents()
+	reagents.add_reagent(generate_type, generate_amount)

```

Comment:
why make a proc that does 1 thing when you could just add the 1 thing to process?